### DISCUSSION TITLE
Does Cadence Web have any authentication mechanisms?

### DISCUSSION THREAD

**Discussion by xgCommander** [uber/cadence#5394] [Q&A] [answered]
Hello team, I have a few questions regarding Cadence Web auth.
Currently, I'm able to directly access Cadence Web UI and view/terminate WFs but I'm wondering if there is a way to do the following:
1. User Access control
2. User Permission management

I couldn't find any documentation about this.

**Comment by Assem-Uber** [comment, marked as answer]
Hello @Vignesh-Ganesan-008, thanks for sharing your questions!
Currently cadence supports JWT auth between Web/Server using 'admin tokens', check the [config section for cadence-web](https://github.com/uber/cadence-web#configuration). This allows the web to be able to communicate with the server if OAuth is enabled. You can enable OAuth and provide public/private for Server and enable admin auth on Web to get auth up and running.
Current limitations are:
- There is no granular access control or permission management on UI
- You need to secure the web domain on your behalf

While we are planning to add more Authentication/Authorization features, the above can serve most common cases.
Finally i will add an issue to add this to the docs, until then let us know if you have any more questions or need more guidance for enabling auth.
